dao/src/main/java/org/wso2/testgrid/dao/repository/TestPlanRepository.java

```diff
@@ -392,6 +392,22 @@ public TestPlan getLastFailure(TestPlan testPlan) {
         return toDelete;
     }
 
+    /**
+     * Delete a list of test_plans from db
+     * @param testPlans list of test plan ids that need to be deleted
+     */
+    public void deleteTestPlans(List<String> testPlans) {
+        String sql;
+        EntityTransaction txn = entityManager.getTransaction();
+        txn.begin();
+
+        for (String testplan : testPlans) {
+            sql = "delete from test_plan where id = '" + testplan + "'";
+            entityManager.createNativeQuery(sql).executeUpdate();
+        }
+        txn.commit();
```

Comment:
Shouldn't this go under a try-catch block? 

It is important because if the line #406 threw an exception, you need to rollback the transaction. ex: https://www.boraji.com/hibernate-5-and-jpa-2-persist-find-merge-and-remove-example